golem-worker-executor-base/src/durable_host/mod.rs

```diff
@@ -299,24 +301,20 @@ impl<Ctx: WorkerCtx> DurableWorkerCtx<Ctx> {
 
     pub async fn store_worker_status(&self, status: WorkerStatus) {
         let oplog_idx = self.state.oplog_idx;
+        let status_record = WorkerStatusRecord {
```

Comment:
The time spent building this (worker status record as a fold over oplog) is proving quite worthwhile, and the architecture for doing new things like durable invocation queue is becoming clearer.

Reply:
Yes I'm very happy about it :)